File: Year-2022/LeetCode-December-Challenge/MinimumAverageDifference.js
var minimumAverageDifference = function (nums) {
    let total = 0, sum = 0;
    for (let x of nums) total += x;
    let mini = Number.MAX_VALUE, res = 0;
    for (let i = 0; i < nums.length; ++i) {
        sum += nums[i];
        let backSum = (total - sum);
        let backSize = (nums.length - i - 1);
        let avg = 0;
        if (backSize == 0) avg = Math.abs(sum / (i + 1));
        else avg = (Math.abs(parseInt(sum / (i + 1)) - parseInt((backSum / backSize))));
        if(mini > avg){
            mini = avg;
            res = i;
        }
    }
    return res;
};
let nums = [0,1,0,1,0,1];
console.log(minimumAverageDifference(nums));

File: Year-2022/LeetCode-December-Challenge/MinimumAverageDifference.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minimumAverageDifference(vector<int>& nums) {
        long long total = 0, sum = 0;
        for(auto ch: nums) total += ch;
        int mini = INT_MAX, res = 0;
        for(int i = 0; i < nums.size(); ++i){
            sum += nums[i];
            long long backSum = total - sum;
            long long backSize = nums.size() - i - 1;
            int avg = 0;
            if(backSize == 0) avg = (sum / (i + 1));
            else avg = (abs((sum)/(i + 1) - (backSum / backSize)));
            if(mini > avg){
                mini = avg;
                res = i;
            } 
        }
        return res;
    }
};
int main(){
    vector<int> nums = {2,5,3,9,5,3};
    Solution *s = new Solution();
    cout<<s->minimumAverageDifference(nums)<<endl;
    return 0;
}